File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        listOfUniques = []
        d = False
        for i in nums:
            for k in listOfUniques:
                if i == k:
                    d = True
            if not d:
                listOfUniques.append(i)
        return d

File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
        vector<int> p;
        vector<vector<int>> out;
        dfs(out, nums, p, target, 0);
        return out;
    }

    void dfs(vector<vector<int>>& out, vector<int>& nums, vector<int>& partition, int curr, int i) {
        if (curr == 0) {
            out.push_back(partition);
            return;
        }
        if (curr < 0 || i >= nums.size()) {
            return;
        }

        curr -= nums[i];
        partition.push_back(nums[i]);
        dfs(out, nums, partition, curr, i);
        curr += nums[i];
        partition.pop_back();
        dfs(out, nums, partition, curr, i + 1);
    }
};
